File: frontend/src/types/image.ts
import { ActType } from "./registry";

/** Editable image metadata */
export interface UserImageMeta {
  name?: string;
  date_range?: string;
  notes?: string;
}

/** Read-only image metadata */
export interface ImageMeta extends UserImageMeta {
  image_number: number;
  act_types: Map<ActType, number>;
  width: number,
  height: number,
  tile_size: number
}

File: backend/plugin_core/src/data/image.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
pub struct Image {
    pub width: Option<u32>,
    pub height: Option<u32>,
    pub tile_size: Option<u32>,
    pub manifest_url: Option<String>,
    pub api_url: Option<String>,
    pub download_url: Option<String>,
    pub ark_url: Option<String>,

    pub image_number: u32,
    pub name: Option<String>,
    pub date_range: Option<String>,
    pub notes: Option<String>,
}
